api/can_api_v2_definition.py

```diff
@@ -0,0 +1,178 @@
+
+
+class MetricsTimeseriesRow(Metrics):
+    """Metrics data for a specific day."""
+
+    date: datetime.date = pydantic.Field(..., descrition="Date of timeseries data point")
+
+
+class RegionSummary(base_model.APIBaseModel):
+    """Summary of actual and prediction data for a single region."""
+
+    fips: str = pydantic.Field(
+        ...,
+        description="Fips Code.  For state level data, 2 characters, for county level data, 5 characters.",
+    )
+    country: str = pydantic.Field(..., description="2-letter ISO-3166 Country code.")
+    state: str = pydantic.Field(..., description="2-letter ANSI state code.")
+    county: str = pydantic.Field(None, description="County name")
+
+    level: AggregationLevel = pydantic.Field(..., description="Level of region.")
+    lat: float = pydantic.Field(None, description="Latitude of point within the state or county")
+    long: float = pydantic.Field(None, description="Longitude of point within the state or county")
+    population: int = pydantic.Field(
+        ..., description="Total Population in geographic region.", gt=0
+    )
+
+    metrics: Metrics = pydantic.Field(None)
+    actuals: Actuals = pydantic.Field(None)
+
+    lastUpdatedDate: datetime.date = pydantic.Field(..., description="Date of latest data")
+
+
+class RegionSummaryWithTimeseries(RegionSummary):
+    """Summary data for a region with prediction timeseries data and actual timeseries data."""
+
+    metricsTimeseries: List[MetricsTimeseriesRow] = pydantic.Field(None)
+    actualsTimeseries: List[ActualsTimeseriesRow] = pydantic.Field(...)
+
+    @property
+    def region_summary(self) -> RegionSummary:
+
+        data = {}
+        # Iterating through self does not force any conversion
+        # https://pydantic-docs.helpmanual.io/usage/exporting_models/#dictmodel-and-iteration
+        for field, value in self:
+            if field not in RegionSummary.__fields__:
+                continue
+            data[field] = value
+
+        return RegionSummary(**data)
+
+
+class AggregateRegionSummary(base_model.APIBaseModel):
+    """Summary data for multiple regions."""
+
+    __root__: List[RegionSummary] = pydantic.Field(...)
+
+    @property
+    def level(self) -> AggregationLevel:
+        return self.__root__[0].level
+
+
+class AggregateRegionSummaryWithTimeseries(base_model.APIBaseModel):
+    """Timeseries and summary data for multiple regions."""
+
+    __root__: List[RegionSummaryWithTimeseries] = pydantic.Field(...)
+
+    @property
+    def level(self) -> AggregationLevel:
+        return self.__root__[0].level
+
+
+class MetricsTimeseriesRowWithHeader(MetricsTimeseriesRow):
+    """Prediction timeseries row with location information."""
+
+    country: str = "US"
+    state: str = pydantic.Field(..., description="The state name")
+    county: str = pydantic.Field(None, description="The county name")
+    fips: str = pydantic.Field(..., description="Fips for State + County. Five character code")
```

Comment:
yes! I imagine that this will eventually be optional and we'll have a separate column that will be location id or similar.  Does that seem reasonable?

Reply:
Yes. We may want to make state optional or repeated too.